essentials-framework/Essentials Core/PepperDashEssentialsBase/Crestron/CrestronGenericBaseDevice.cs

```diff
@@ -69,19 +69,28 @@ protected void RegisterCrestronGenericBase(GenericBase hardware)
 		public override bool CustomActivate()
 		{
             Debug.Console(0, this, "Activating");
-			if (!PreventRegistration)
-			{
+            if (!PreventRegistration)
+            {
                 //Debug.Console(1, this, "  Does not require registration. Skipping");
 
-				var response = Hardware.RegisterWithLogging(Key);
-				if (response != eDeviceRegistrationUnRegistrationResponse.Success)
-				{
-					//Debug.Console(0, this, "ERROR: Cannot register Crestron device: {0}", response);
-					return false;
-				}
+                var response = Hardware.RegisterWithLogging(Key);
+                if (response != eDeviceRegistrationUnRegistrationResponse.Success)
+                {
+                    //Debug.Console(0, this, "ERROR: Cannot register Crestron device: {0}", response);
+                    return false;
+                }
 
                 IsRegistered.FireUpdate();
-			}
+            }
+            else
+            {
+                AddPostActivationAction(() =>

```

Comment:
The question I have here is what should the `PreventRegistration` flag be used for? Should it be used to prevent automatic registration or just delay it until a different activation phase?

Maybe there should be a second flag called `DelayRegistration` that has this behavior, and `PreventRegistration` would be used to prevent automatic registration at all? Maybe with XML comments that indicate what the behavior is for each one?